packages/fxa-customs-server/lib/dataflow.js

```diff
@@ -72,12 +81,16 @@ module.exports = (config, log, fetchRecords, setRecords) => {
           [type]: action.indicator,
         });
 
-        records[type][action.suggested_action]();
-
-        await setRecords(records);
+        if (recordType) {
+          records[recordType][action.suggested_action]();
```

Comment:
This was undefined before because it was doing records['email'].block(), it should be records['emailRecord'].block()